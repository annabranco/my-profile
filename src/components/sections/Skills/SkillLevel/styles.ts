import styled from 'styled-components';
import {
  colorBlack,
  colorWhite,
  colorYellowPale
} from '../../../../styles/theme';

export const Star = styled.i`
  border-radius: 50%;
  background: ${colorBlack};
  height: 20px;
  width: 20px;
  padding-top: 5px;
  text-shadow: 0 0 5px ${colorBlack};
  font-size: 0.8rem;
  color: ${colorYellowPale};
  box-shadow: 0 0 5px 2px ${colorWhite};
  text-align: center;
`;
Star.displayName = 'Star';

export const StarsWrapper = styled.div`
  position: absolute;
  opacity: 0;
  width: 100%;
  margin-bottom: 10px;
  text-align: center;
  transition: opacity 0.5s ease;

  &:hover {
    opacity: 0.9;
  }
`;
StarsWrapper.displayName = 'Stars Wrapper';
