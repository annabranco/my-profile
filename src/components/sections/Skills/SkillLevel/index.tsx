import { ReactElement } from 'react';
import { number, string } from 'prop-types';
import { Star, StarsWrapper } from './styles';

interface Props {
  level: number;
  skill: string;
}

const SkillLevel = ({ level, skill }: Props): ReactElement => {
  const stars: '*'[] = new Array(level).fill('*');

  return (
    <StarsWrapper data-e2e-id={`skill-${skill}_starsWrapper`}>
      {stars.map((_, index) => (
        <Star
          aria-hidden
          data-e2e-id={`skill-${skill}_star`}
          className="fas fa-star"
          key={`${skill}-${level}-${index}`} // eslint-disable-line react/no-array-index-key
        />
      ))}
    </StarsWrapper>
  );
};

SkillLevel.propTypes = {
  skill: string.isRequired,
  level: number.isRequired
};

export default SkillLevel;
