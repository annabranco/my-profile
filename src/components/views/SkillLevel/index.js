import React from 'react';
import { number } from 'prop-types';
import { StarsWrapper, Star } from './styles';

const SkillLevel = ({ level }) => {
  const stars = new Array(level).fill('*');

  return (
    <StarsWrapper>
      {stars.map(() => (
        <Star className="fas fa-star" aria-hidden key={Math.random()} />
      ))}
    </StarsWrapper>
  );
};

SkillLevel.propTypes = {
  level: number
};

SkillLevel.defaultProps = {
  level: 1
};

export default SkillLevel;
