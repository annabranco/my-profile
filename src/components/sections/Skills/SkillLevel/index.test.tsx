import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import SkillLevel from '.';
import { setupMockProvider } from '../../../../../testing/mocks';

// TODO [07-May -21]: Define real test suits (Anna Branco)

describe('< SkillLevel >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <SkillLevel level={3} skill="mockSkill" />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
