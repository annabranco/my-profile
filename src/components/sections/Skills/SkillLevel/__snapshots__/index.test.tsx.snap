// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< SkillLevel > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SkillLevel
    level={3}
    skill="mockSkill"
  />
</Provider>
`;
